crowbar_framework/app/models/deployer_service.rb

```diff
@@ -77,7 +77,8 @@ def transition(inst, name, state)
     if state == "hardware-installing"
       node = NodeObject.find_node_by_name(name)
       # build a list of current and pending roles to check against
-      roles = node.roles.dup
+      roles = []
+      roles = node.roles.dup if node.roles
```

Comment:
i knew there was a more elegant way to do it, but i'm no ruby expert and that's the best I could came up with.

If the change works I'll submit a PR with that syntax.